File: DesignPatternsDotNet/VisitorPattern/Fruit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace VisitorPattern
{
    class Fruit
    {
        private double price;
        public Fruit(double aPRice)
        {
            price = aPRice;
        }
        public double Price
        {
            get
            {
                return price;
            }
        }
        public double AddTaxMonitor(ITaxMonitor taxMonitor)
        {
            return taxMonitor.AddTax(this);
        }
    }
}

File: DesignPatternsJava/src/main/java/com/visitor/Fruit.java
package com.visitor;

public class Fruit implements ITaxable{
	private double price;
	public Fruit(double aPRice){
		price=aPRice;
	}public double getPrice(){
		return price;
	}
	public double addTaxMonitor(ITaxMonitor taxMonitor) {		
		return taxMonitor.addTax(this);
	}
}
